src/matches2.js

```diff
@@ -0,0 +1,44 @@
+import {
+  all, compose, equals, mapObjIndexed, values, path, is, flatten, cond, map, identity, T, addIndex,
+} from 'ramda';
+
+const mapIndexed = addIndex(map);
+
+const recursivelyFlatten = (value) => cond([
+  [is(Array), compose(map(recursivelyFlatten), flatten)],
+  [T, identity],
+])(value);
+
+const mapObjectToMatch = (objectToMatch, currentPath = []) => mapObjIndexed(
+  (value, key) => cond([
+    [
+      is(Object),
+      compose(values, mapObjectToMatch(objectToMatch, [...currentPath, key])),
+    ],
+    [
+      is(Array),
```

Comment:
Je crois qu'un array est également un objet, et si on fait le pattern matching dans cet ordre un array risque de passer dans le case object.

Reply:
Oui, il faudrait faire sinon `and(is(Object), not(is(Array.isArray))` peut-être non?

Pour ça que dans ma version j'avais juste fais un `if(typeof 'object')` qui couvrait les deux cas de figure du coup.

Reply:
Ou simplement mettre le cas is(Array) avant le is(Object)